src/lib/cozy-client/slices/sharings.js

```diff
@@ -238,6 +238,38 @@ const createSharingLink = document => ({
   promise: client => client.createSharingLink(getPermissionsFor(document, true))
 })
 
+// TODO: this is a poor man's migration in order to normalize contacts
+// and should be removed after a few weeks in prod
+export const fetchContacts = () => {
+  const action = fetchCollection('contacts', 'io.cozy.contacts')
+  action.promise = async client => {
+    const response = await client.fetchCollection(
+      'contacts',
+      'io.cozy.contacts'
+    )
+    const data = await Promise.all(
+      response.data.map(contact => {
+        return typeof contact !== 'string'
+          ? contact
+          : client
+              .updateDocument({
+                ...contact,
+                email: [
+                  {
+                    address: contact.email,
+                    primary: true
+                  }
+                ]
+              })
+              .then(resp => resp.data[0])
```

Comment:
I think we should avoid mixing `async/await` and `then()`, if possible.

Reply:
I agree, but in that case it would mean a less FP-oriented style

Reply:
Well, it would still be FP. Something like this :

```
response.data.map(async (contact) => {
  if (typeof contact !== 'string') {
    return contact
  } else {
    const resp = await client.updateDocument(…)
    return resp.data[0]
  }
})
```

I know it gets rid of the ternary which is quite elegant here, but I think the more verbose version adds some clarity. It becomes obvious that what is returned is `contact` or `return resp.data[0]`, instead of `contact` or a promise that resolves.

But I can live with both! It's easy to compare both versions now so pick whatever you feel is best :)

Reply:
I have to admit that current code is not very clear.